### DISCUSSION TITLE
[Feature Request] Colour Picker

### DISCUSSION THREAD

**Discussion by shuvashish76** [VishnuSanal/Quotes#96] [Ideas]
Its a complete new feature request & not related to  app accent colour (#19 & #20)

Here we pick pick background/font/card colour from colour picker. Plus pick font/card colour from quote app background itself. Only "**A**" for background, "**A+B**" for font/card

So there are 2 types of colour picker
**A**= https://github.com/skydoves/ColorPickerView/blob/master/art/art0.gif (basic colour picker)
**B**= https://github.com/skydoves/ColorPickerView/blob/master/art/art1.gif (picker from image)
Source: [ColorPickerView](https://github.com/skydoves/ColorPickerView) by skydoves

Both A & B gifs are same just the circle change to a different image

### Design:
version1
![IMG_20210809_153810](https://user-images.githubusercontent.com/48226312/128778262-07683fd4-3882-4e55-b94a-b6cfa2c52690.jpg)

everything is self-explanatory here.

"**Live quote-image preview**" is a new feature request. Basically it show preview of final quote image on live as you change background/font/card colour #27.

---
Another thing I changed is "**default colour show option**"

Current version: (slide right to view more default colours)
![Screenshot_2021-08-10-03-28-45](https://user-images.githubusercontent.com/48226312/128779857-aa0e97cd-9228-49a8-a28e-b323e7f196d4.jpeg)

My suggestion: (all default colours in 1 page)
![Screenshot_2021-08-10-03-31-36](https://user-images.githubusercontent.com/48226312/128780125-25d295c2-b30a-427a-b9e0-bdc7b959c3d1.jpeg)
or
![Screenshot_2021-08-10-03-31-54](https://user-images.githubusercontent.com/48226312/128780168-fd27804b-0aee-4027-b0ae-51f44311d368.jpeg)

screenshort source: [ColorPicker](https://github.com/Dhaval2404/ColorPicker/releases) by Dhaval2404
(I prefer skydoves' project since it has option to pick colour brightness & saturation options from a preferred image)

Thankyou🙏🏿

**Comment by shuvashish76** [comment]
@VishnuSanal from our discussion it seems +gallery & recent options are unnecessary.

 ### Design
version2
![IMG_20210810_044934_708](https://user-images.githubusercontent.com/48226312/128786194-3dd6e673-52d6-4166-a280-f3b6e1dda23d.jpg)

What we changed.
1. Remove all those upper buttons/options
2. Slide the circle horizontally left to view **B** (picker from background image)
3. Picker option (whether its the circle or image) should be relatively bigger than preview option, as we need more space to pick colour accurately from the circle/image.
4. That slide left option is not necessary for background it is only for font & card
(Why? Check 2nd line of my previous comment)
> Here we pick pick background/font/card colour from colour picker. Plus pick font/card colour from quote app background itself. Only "**A**" for background, "**A+B**" for font/card

**Comment by shuvashish76** [comment]
5. Remember picker methord
(Option to rememeber if we used **A** (colour picker) or **B** (picker from image) last time... Useful each time user change background but doesn't want to see A option 1st as always & then swipe left for B option while picking font/card colour.

6. Pin recent colour
![Screenshot_2021-08-10-15-15-59](https://user-images.githubusercontent.com/48226312/128853246-e31b1a98-5d31-4777-8e9b-de5abc03476e.jpeg)

(is it necessary as we've recent option already?
![Screenshot_2021-08-10-15-28-32](https://user-images.githubusercontent.com/48226312/128853601-20359f52-08b4-4a71-9b77-5bd3a4fde67c.jpeg)

Lets say I really love this colour & want mostly use this unique colour for font/card even if I use a different background.
i. Its not available on default colours
ii. Cant choose from background (**B**) as set image is a different one
iii. Tough to find the exact matching colour from the circle picker (**A**)
iv. Not available on recent colours 

So solution is simply pin that colour :D)

**Comment by shuvashish76** [comment]
I think it would be better to convert this issue to a discussion 1st.
Then break into smaller issues & add each feature on top of main feature* one by one
- picker issue (main issue)
- preview issue
- pick from background image
- recent colours
- pin recents
etc...

**Comment by VishnuSanal** [comment]
> I think it would be better to convert this issue to a discussion 1st. Then break into smaller issues & add each feature on top of main feature* one by one

Yess! Please confirm the following.

* picker issue (main issue)

> Here we pick pick background/font/card colour from colour picker. 

Did you mean to say this?

* preview issue

Live preview whik

> "Live quote-image preview" is a new feature request. Basically it show preview of final quote image on live as you change background/font/card colour 

* pick from background image

> The feature referred to with "B". https://github.com/skydoves/ColorPickerView/blob/master/art/art1.gif (picker from image)

 * recent colours
 * pin recents

:+1:

I vote for these too (these can be included along with the main issue)

> (all default colours in 1 page)

**Reply by VishnuSanal** [reply]
@shuvashish76 Please confirm my assumptions. We can move these to individual issues after our discussion :)

**Reply by shuvashish76** [reply]
- _picker issue (main issue)_ = yes, new design for picker in general (for now [basic picker](https://github.com/skydoves/ColorPickerView/blob/master/art/art0.gif))
- 👍🏿
- 👍🏿

**Comment by VishnuSanal** [comment]
> Then break into smaller issues & add each feature on top of main feature* one by one

- picker issue (main issue)

 fixed via #107

- preview issue

 tracked via #120

- pick from background image

 tracked via #121

- recent colours

 tracked via #122

- pin recents

 tracked via #123

**Reply by VishnuSanal** [reply]
@shuvashish76 please verify the above & thanks for your suggestions!! :)

closing this discussion as every suggestion mentioned here are tracked via individual issues now.
